### DISCUSSION TITLE
GitFlow with release branches: Gitversion does not generate correct version

### DISCUSSION THREAD

**Discussion by dbelyaev** [GitTools/GitVersion#4068] [Q&A]
Hello!

I am trying to set up a GitHub Action pipeline that supports the GitFlow branching strategy and want to use GitVersion to automate version generation and tagging based on branch names and existing tags.

However, it seems like I am having issues or cannot figure out how to configure GitVersion correctly for my use case.

I have just added a public repository that shows my issues: https://github.com/dbelyaev/gitversion-and-gitflow

The flow is simple, we have two branches:

* `develop`
* `main`

Where the pipeline looks like this:

* On push to develop

  * Auto-deploy to DEV environment
  * No tagging needed

* On new pull request (opened, synchronize, reopened) from `release/vX.X.X` or `hotfix/vX.X.X` to `main`

  * Auto-deploy to TEST environment
  * Figure out the new version with GitVersion from the branch name (the part `vX.X.X`)
  * Add `rc` suffix to get something like `vX.X.X-rcY`
  * Use this version to tag the latest commit on the branch

* On pull request to `main` close and merged

  * Auto-deploy to PROD environment

* On push to `main`

  * Open a back-merge PR targeting `develop` to sync branches
  * Figure out the new release version with GitVersion (from previous `rc` tags and merge commit message with `Merge of release/vX.X.X`)
  * Version looks like `vX.X.X`
  * Use this version to tag the latest commit on the `main` branch

## Current issue

Looks like gitversion is not calculating version using release branch name, but activate pull-request related versioning:

* I already have a release tag on main: `v0.0.1`
* I have added a new feature to `develop` and started a new release branch `release/0.2.4`
* I have added a [PR](https://github.com/dbelyaev/gitversion-and-gitflow/pull/6) from `release/0.2.4` targeting `main`
* GitVersion run on this PR determined semver to be `0.1.0-PullRequest0006.9` while I expect it to be `0.2.4-rc1` (see step "Display Semver" in [this job output](https://github.com/dbelyaev/gitversion-and-gitflow/actions/runs/9513849706/job/26224784775?pr=6))

So, I expect gitversion to use `Version in branch name` from original release branch as source and not an internal GitHub pr related branch name.

My [gitVersion.yml](https://github.com/dbelyaev/gitversion-and-gitflow/blob/develop/gitVersion.yml) looks like this:

```yml
branches:
  develop:
    tag: beta
  release:
    tag: rc
  hotfix:
    tag: rc
```

Any help on how to correctly configure gitversion?

**Comment by HHobeck** [comment]
It's not clear for me what you are trying to do. Generally starting a discussion like **Gitversion does not generate correct version**  with docent of use cases is not so ideal. Please isolate your problem and create a separate discussion or issue.

Anyway I would recommend you to migrate to the preview version 6.0.0-rc.2 and start using the supported `GitFlow `workflow.

[GitFlow examples -> Create release branch with version](https://github.com/GitTools/GitVersion/blob/main/docs/input/docs/learn/branching-strategies/gitflow/examples.md#release-branches)

![image](https://github.com/user-attachments/assets/b3bfe80c-780a-422c-b93c-8dcfba647395)
(Link:  [DocumentationSamplesForGitFlow_VersionedReleaseBranch](https://github.com/GitTools/GitVersion/blob/main/docs/input/docs/img/DocumentationSamplesForGitFlow_VersionedReleaseBranch.png))

**Comment by dbelyaev** [comment]
Latest v6 resolved the issue.
